daft/__init__.py

```diff
@@ -98,6 +99,7 @@ def get_build_type() -> str:
     "read_iceberg",
     "read_delta_lake",
     "read_sql",
+    "read_lance",
```

Comment:
We also should have the lance feature in the pyproject.toml to install the correct deps